### PR TITLE
fix: Fix broken archive UI Fixes #10606

### LINKED ISSUE #10606: Archive UI broken since v3.4.5
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [x] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

Screenshot 2023-02-24 at 12 57 32

Screenshot 2023-02-24 at 12 57 55

Screenshot 2023-02-24 at 12 58 22

NOTE: v3.4.4 does not have the issue

### Version

v3.4.5, master (786639d4e1bb279894e4f36388f83b721990b261)

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
not related to particular workflow, but Archived Workflows UI component. Apparently regression after v3.4.4.
```

### Logs from the workflow controller

```text
No errors
```

### Logs from in your workflow's wait container

```text
N/A (not related to workflow execution)
```

### PR DESCRIPTION
**Pull request by Kitanotori**
Fixes #10606

### CODE REVIEW

**Comment by Kitanotori** [comment]
How can we prevent dependabot from breaking things?
#10547
#10575
#10587

**Comment by Kitanotori** [comment]
@juliev0 I don't quite understand the failed check. I was unable to run `make pre-commit -B` at local with following error:

```
github.com/argoproj/argo-workflows/pkg/apis/workflow/v1alpha1/generated.proto: File not found.
pkg/apiclient/clusterworkflowtemplate/cluster-workflow-template.proto:6:1: Import "github.com/argoproj/argo-workflows/pkg/apis/workflow/v1alpha1/generated.proto" was not found or had errors.
pkg/apiclient/clusterworkflowtemplate/cluster-workflow-template.proto:14:3: "github.com.argoproj.argo_workflows.v3.pkg.apis.workflow.v1alpha1.ClusterWorkflowTemplate" is not defined.
```

File: ui/src/app/shared/services/archived-workflows-service.ts [hunk 1]
```diff
@@ -39,12 +39,12 @@ export class ArchivedWorkflowsService {
...
-            return requests.get(`api/v1/archived-workflows-label-values?listOptions.labelSelector=${key}`).then(res => res.body as models.Labels);
-        } else {
-            return requests.get(`api/v1/archived-workflows-label-values?namespace=${namespace}&listOptions.labelSelector=${key}`).then(res => res.body as models.Labels);
+    public async listLabelValues(key: string, namespace: string): Promise<models.Labels> {
+        let url = `api/v1/archived-workflows-label-values?listOptions.labelSelector=${key}`;
+        if (namespace !== '') {
+            url += `&namespace=${namespace}`;
         }
+        return (await requests.get(url)).body as models.Labels;
     }
```
Comment (juliev0) [inline ui/src/app/shared/services/archived-workflows-service.ts:48]: that's cleaner :)

Comment (Kitanotori) [reply ui/src/app/shared/services/archived-workflows-service.ts:48]: Aa, I see. Just felt that this
```ts
    private async fetchArchivedWorkflowsLabels(key: string): Promise<string[]> {
        const labels = await services.archivedWorkflows.listLabelValues(key, this.props?.namespace);
        return labels.items
            .map(i => key + '=' + i)
            .sort((a, b) => a.localeCompare(b));
    }
```
is easier to read than this
```ts
    private async fetchArchivedWorkflowsLabels(key: string): Promise<string[]> {
        const labels = await services.archivedWorkflows.listLabelValues(key, this.props?.namespace);
        return labels.items.map(i => key + '=' + i).sort((a, b) => a.localeCompare(b));
    }
```
I guess the linter disagrees ^^;

Comment (juliev0) [reply ui/src/app/shared/services/archived-workflows-service.ts:48]: I’m with you there…go figure

File: ui/package.json [hunk 2]
```diff
@@ -57,7 +57,7 @@
         "@types/react-dom": "^16.8.2",
         "@types/react-form": "^2.16.1",
         "@types/react-helmet": "^6.1.6",
-        "@types/react-router-dom": "^5.3.3",
+        "@types/react-router-dom": "^4.2.3",
         "@types/superagent": "^4.1.16",
```
Comment (juliev0) [inline ui/package.json:61]: curious what the issue was with dependabot updating these?

Comment (Kitanotori) [reply ui/package.json:61]: That update broke the whole UI. Was surprised that such changes can be merged without testing..

Comment (juliev0) [reply ui/package.json:61]: Wow, thanks for fixing that. I guess we need better automated tests for the UI.

Comment (juliev0) [reply ui/package.json:61]: ahh, now I see all 3 of the changes you referenced: dependabot broke it, somebody reverted that, and dependabot did the same thing again afterward. I'll bring it up tomorrow when I meet with one of the leads.

**Review by juliev0 — APPROVED** [review]

**Review by alexec — APPROVED** [review]
Julie approves, so I approve.

**Pull request merged**